core/shared/src/test/scala/fs2/StreamSuite.scala

```diff
@@ -1031,6 +1031,13 @@ class StreamSuite extends Fs2Suite {
     assertEquals(countChunks(source), 2)
   }
 
+  test("rechunkRandomly sometimes emits everything in a single chunk") {
+    val fiveChunks = Stream(1) ++ Stream(2) ++ Stream(3) ++ Stream(4) ++ Stream(5)
+
+    val source = fiveChunks.rechunkRandomlyWithSeed(0.1, 2.0)(5).chunks.toList
+    assert(source.forall(_.size <= 2), "Some Chunks have larger size than maxFactor")
```

Comment:
To clarify: what `rechunkRandomly` does is to allow the output to have chunks of size up to `chunkSize * maxFactor`, so if the input were chunks of size 2, it should allow output chunks of size up to 4.

Reply:
Oh, I ignore that in the comment. Thanks for that.